Lib/profiling/sampling/_child_monitor.py

```diff
@@ -0,0 +1,279 @@
+            cli_args: CLI arguments to pass to child profilers
+            output_pattern: Pattern for output files (format string with {pid})
+        """
+        self.parent_pid = pid
+        self.cli_args = cli_args
+        self.output_pattern = output_pattern
+
+        self._known_children = set()
+        self._spawned_profilers = []
+        self._lock = threading.Lock()
+        self._stop_event = threading.Event()
+        self._monitor_thread = None
+        self._poll_count = 0
+
+    def __enter__(self):
+        self._monitor_thread = threading.Thread(
+            target=self._monitor_loop,
+            daemon=True,
+            name=f"child-monitor-{self.parent_pid}",
+        )
+        self._monitor_thread.start()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self._stop_event.set()
+        if self._monitor_thread is not None:
+            self._monitor_thread.join(timeout=2.0)
+            if self._monitor_thread.is_alive():
+                print(
+                    "Warning: Monitor thread did not stop cleanly",
+                    file=sys.stderr,
+                )
+
+        # Wait for child profilers to complete naturally
+        self.wait_for_profilers()
+
+        # Terminate any remaining profilers
+        with self._lock:
+            profilers_to_cleanup = list(self._spawned_profilers)
+            self._spawned_profilers.clear()
+
+        for proc in profilers_to_cleanup:
+            self._cleanup_process(proc)
+        return False
+
+    def _cleanup_process(self, proc, terminate_timeout=2.0, kill_timeout=1.0):
+        if proc.poll() is not None:
+            return  # Already terminated
+
+        proc.terminate()
+        try:
+            proc.wait(timeout=terminate_timeout)
+        except subprocess.TimeoutExpired:
+            proc.kill()
+            try:
+                proc.wait(timeout=kill_timeout)
+            except subprocess.TimeoutExpired:
+                # Last resort: wait indefinitely to avoid zombie
+                # SIGKILL should always work, but we must reap the process
+                try:
+                    proc.wait()
+                except Exception:
+                    pass
+
+    @property
+    def spawned_profilers(self):
+        with self._lock:
+            return list(self._spawned_profilers)
+
+    def wait_for_profilers(self, timeout=_DEFAULT_WAIT_TIMEOUT):
+        """
+        Wait for all spawned child profilers to complete.
+
+        Call this before exiting the context if you want profilers to finish
+        their work naturally rather than being terminated.
+
+        Args:
+            timeout: Maximum time to wait in seconds
+        """
+        profilers = self.spawned_profilers
+        if not profilers:
+            return
+
+        print(
+            f"Waiting for {len(profilers)} child profiler(s) to complete...",
+            file=sys.stderr,
+        )
+
+        deadline = time.time() + timeout
```

Comment:
```suggestion
        deadline = time.monotonic() + timeout
```

Maybe use monotonic time?

Reply:
Ah good catch!